### ISSUE TITLE
Compiling classes on the fly in Windows 8.1

### ISSUE THREAD

**Issue by metall0id** [mwrlabs/drozer#130]
Reported by Caitlin Harrison.

The error in question is:

```
dz> run app.test.greeting
"C:\Program Files (x86)\Java\jdk1.6.0_21\bin\javac.exe" -cp C:\drozer\Lib\drozer\lib\android.jar HelloWorld.java
exception in module: OSError: [Errno 22] Invalid argument
Traceback (most recent call last):
  File "C:\drozer\Lib\drozer\console\session.py", line 394, in do_run
    module.run(argv[1:])
  File "C:\drozer\Lib\drozer\modules\base.py", line 183, in run
    result = self.execute(arguments)
  File "c:\drozer\Lib\drozer\modules\app\test\HelloWorld.py", line 21, in execute
    hello = self.loadClass("app/test/HelloWorld.apk", "C:/drozer/Lib/drozer/modules/app/test/HelloWorld")
  File "C:\drozer\Lib\drozer\modules\common\loader.py", line 30, in loadClass
    Module.cache_klass(".".join([source, klass]), loader.loadClass(klass))
  File "C:\drozer\Lib\pydiesel\reflection\utils\class_loader.py", line 28, in loadClass
    return self.getClassLoader().loadClass(klass);
  File "C:\drozer\Lib\pydiesel\reflection\utils\class_loader.py", line 36, in getClassLoader
    self.source = self.__get_source(self.source_or_relative_path, relative_to=self.relative_to)
  File "C:\drozer\Lib\pydiesel\reflection\utils\class_loader.py", line 80, in __get_source
    source = ClassBuilder(java_path, self.dx_path(), self.javac_path(), self.android_path()).build()

  File "C:\drozer\Lib\pydiesel\reflection\utils\class_builder.py", line 36, in build
    if not self.__execute(self.javac , "-cp", self.sdk_path, os.path.basename(self.path)):
  File "C:\drozer\Lib\pydiesel\reflection\utils\class_builder.py", line 68, in __execute
    return os.spawnve(os.P_WAIT, argv[0], argv, os.environ) == 0
OSError: [Errno 22] Invalid argument
```

This was solved by manually compiling the apk, using the following two commands:

```
C:\Users\Caitlin>javac -cp C:\drozer\Lib\drozer\lib\android.jar C:\drozer\Lib\drozer\modules\app\test\HelloWorld.java
C:\Users\Caitlin>"C:\Android\android-sdk\platforms\android-4\tools\dx.bat" --dex --no-strict --output=C:\drozer\Lib\drozer\modules\app\test\HelloWorld.apk C:\drozer\Lib\drozer\modules\app\test\HelloWorld.class
```

If the `--no-strict` option isn’t used with `dx`, I get a path does not match path error. Windows throws its toys out the cot if full paths aren't used everywhere.

System details for reference are as follows:

```
C:\Users\Caitlin>java -version
java version "1.6.0_21"
Java(TM) SE Runtime Environment (build 1.6.0_21-b07)
Java HotSpot(TM) Client VM (build 17.0-b17, mixed mode, sharing)

Windows 8.1 Pro Version 6.3 (Build 9600)
```

**Comment by raulsiles** [comment]
Definitely, the bug is in the os.spawnve() function:
...
return os.spawnve(os.P_WAIT, argv[0], argv, os.environ) == 0
OSError: [Errno 22] Invalid argument

The same scenario occurs in Windows 7 64-bit when running the following Drozer commands (as a sample):
dz> run app.package.manifest com.mwr.example.sieve
or
dz> run scanner.provider.finduris -a com.mwr.example.sieve
(Issue #165 is related to this same error)

Manually compiling any of the associated APKs with the previously indicated two commands fixes the issue.

**Comment by amarekano** [comment]
Fixed with PR #299.
